### ISSUE TITLE
Segmentation fault seemingly at random with v1.2.1

### ISSUE THREAD

**Issue by edgardomortiz** [smithlabcode/falco#55]
I have analyzed this file successfully in the past with previous versions of Falco, but now version 1.2.1 is producing a Segmentation fault (identical behavior in Linux and Mac)

```
$ falco --outdir test_falco --threads 1 Antrev_R2.fq.gz
[Mon Oct  9 13:13:27 2023] creating directory for output: test_falco
[limits]	using file /Users/emortiz/y/envs/captus/opt/falco/Configuration/limits.txt
[adapters]	using file /Users/emortiz/y/envs/captus/opt/falco/Configuration/adapter_list.txt
[contaminants]	using file /Users/emortiz/y/envs/captus/opt/falco/Configuration/contaminant_list.txt
[Mon Oct  9 13:13:27 2023] Started reading file Antrev_R2.fq.gz
[Mon Oct  9 13:13:27 2023] reading file as gzipped FASTQ format
[running falco|===================================================|100%]
[Mon Oct  9 13:13:27 2023] Finished reading file
[Mon Oct  9 13:13:27 2023] Writing summary to test_falco/summary.txt
[Mon Oct  9 13:13:27 2023] Writing text report to test_falco/fastqc_data.txt
[Mon Oct  9 13:13:27 2023] Writing HTML report to test_falco/fastqc_report.html
Segmentation fault: 11
```
I hope you can help... I am also attaching the reads
[Antrev_R2.fq.gz](https://github.com/smithlabcode/falco/files/12845304/Antrev_R2.fq.gz)

Edgardo

**Comment by edgardomortiz** [comment]
Falco was installed with bioconda, the corresponding _R1 on this sample works fine.

**Comment by xuxiaodong111** [comment]
Yes, I met the same problem, too.When run in batches, R1 is fine, but some R2 files failed: Segmentation fault      (core dumped) falco. And I use Falco 1.2.1 (bioconda), too.

**Comment by edgardomortiz** [comment]
Yeah, I guess we should declare Falco as abandonware... seems that is hasn't been maintained since 2022

**Comment by yangli04** [comment]
I also reproduced your error in falco 1.2.1 if I use your .gz file. This problem might be caused by gzip methods. I unzipped your file @edgardomortiz and it passed using falco 1.2.1.

Then if I use the newest version of falco (v1.2.2) it will not have any problem. This problem was also mentioned here: 
https://github.com/smithlabcode/falco/issues/45

**Comment by andrewdavidsmith** [comment]
Thanks for doing that @yangli04. If you are able to close this issue, the next release might automatically credit you; otherwise I'll try to remember to credit you for your help.

@edgardomortiz Sorry that I missed this issue when first opened. I have not abandoned Falco. In fact, I was thinking about it this morning before arriving at work. But I can see why it might have appeared so.

**Comment by yangli04** [comment]
I think the issue is solved in v1.2.2, although bioconda still gives you v1.2.1 by default. Also, I cannot close issue here since I am not owner or collaborator. 😂

**Comment by andrewdavidsmith** [comment]
@yangli04 Thanks for pointing out the issue with conda not favoring 1.2.2 yet. I'll see what I can do about that. It's been a frustrating kind of issue in the past. I'll close this one, and if you have any interest in getting involved, my email should be easy to find.
